lib/media/streaming_engine.js

```diff
@@ -1271,7 +1277,30 @@ shaka.media.StreamingEngine = class {
           }
         };
 
-        await this.fetch_(mediaState, reference, streamDataCallback);
+        const result =
+            await this.fetch_(mediaState, reference, streamDataCallback);
+        if (!callbackCalled) {
```

Comment:
This line could use a more comprehensive comment.
Something like...

```js
// In some environments, we might be forced to use network plugins that don't support
// streamDataCallback. In those cases, as a fallback, append the buffer here.
```

Reply:
Done!